src/neo/Ledger/TransactionVerificationContext.cs

```diff
@@ -13,25 +13,48 @@ public class TransactionVerificationContext
         /// </summary>
         private readonly Dictionary<UInt160, BigInteger> senderFee = new Dictionary<UInt160, BigInteger>();
 
+        /// <summary>
+        /// Store oracle responses
+        /// </summary>
+        private readonly Dictionary<ulong, UInt256> oracleResponses = new Dictionary<ulong, UInt256>();
+
         public void AddTransaction(Transaction tx)
         {
             if (senderFee.TryGetValue(tx.Sender, out var value))
                 senderFee[tx.Sender] = value + tx.SystemFee + tx.NetworkFee;
             else
                 senderFee.Add(tx.Sender, tx.SystemFee + tx.NetworkFee);
+
+            var oracle = tx.GetAttribute<OracleResponse>();
+            if (oracle != null) oracleResponses.TryAdd(oracle.Id, tx.Hash);
```

Comment:
If the id already exists in `oracleResponses`, then the transaction should not be added, right?

Reply:
Please take a look, Remove was designed to always remove, so I changed it a little.

Reply:
I found that before adding a transaction, `CheckTransaction()` is always called first, so it will not fail.

Reply:
Reverted